src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
 'use strict';
 
+const DEFAULT_HEALTH = 100;
+const BITE = 50;
+
 class Animal {
-  // write your code here
+  constructor(name, health = DEFAULT_HEALTH) {
+    this.name = name;
+    this.health = health;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    }
+
+    Animal.alive.push(this);
+  }
+
+  die() {
+    if (Animal.alive) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index !== -1) {
+        Animal.alive.splice(index, 1);
+      }
```

Comment:
[CODE PERFOMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead